guacamole/src/main/frontend/src/app/rest/services/tunnelService.js

```diff
@@ -96,20 +96,18 @@ angular.module('rest').factory('tunnelService', ['$injector',
      *     success.
      */
     service.getProtocol = function getProtocol(tunnel) {
-
-        // Build HTTP parameters set
-        var httpParameters = {
-            token : authenticationService.getCurrentToken()
-        };
-
+        // GUACAMOLE-956: replace http parameter with header | START
         // Retrieve the protocol details of the specified tunnel
         return requestService({
             method  : 'GET',
             url     : 'api/session/tunnels/' + encodeURIComponent(tunnel)
                         + '/protocol',
-            params  : httpParameters
-        });
+            headers: {
+                'Guacamole-Token': authenticationService.getCurrentToken(),
+            }
```

Comment:
Instead of manually including the `Guacamole-Token` header, it would be better to call `authenticationService.request()` which adds the header automatically. For example:

https://github.com/apache/guacamole-client/blob/d9bb38eb65dd7e2f5fd1b13fa108d9d19c7cab4a/guacamole/src/main/frontend/src/app/rest/services/tunnelService.js#L78-L81

Reply:
@mike-jumper Thank you so much for all the responses and tips. I have created a Jira for this. Will submit a new PR by following the suggestions. 

Best,
Ares

Reply:
Adding the Jira here: https://issues.apache.org/jira/browse/GUACAMOLE-1775